### ISSUE TITLE
Add support for jupyter notebook ".ipynb" files

### ISSUE THREAD

**Issue by dev4min** [Gruntfuggly/todo-tree#556]
Hi, thanks for your great work, this extension is really helpful. 🙏
Unfortunately it doesn't work with `.ipynb` files as expected.
Some issues are:

- TAGs detection not working, until file get saved and todo-tree refreshed manually ↻
- Doesn't jump to TAG line
- TAGs highlight not working correctly

**Comment by Gruntfuggly** [comment]
Can you try adding `vscode-notebook-cell` to the `todo-tree.general.schemes` setting.

Please have a look at this issue and see if it helps:
https://github.com/Gruntfuggly/todo-tree/issues/548

**Comment by dev4min** [comment]
> Can you try adding `vscode-notebook-cell` to the `todo-tree.general.schemes` setting.

It worked like a charm, thank you 🙏
It'll be great if it be enabled by default for new users who don't know much about this settings.

Now just one thing is getting in the way, it just detect active cell TAGs not whole the notebook.
In `Scan mode: workspace only` it shows all TAGs but can't jump to related cell.

![image](https://user-images.githubusercontent.com/89294258/135050822-a9030df9-3ad1-4eea-8b56-a54c73cd878c.png)

**Comment by Gruntfuggly** [comment]
I'll update the default value for the `todo-tree.general.schemes` setting.

Could you post your settings and an example notebook so I could try it out? I don't really know much about notebooks I'm afraid.

**Comment by dev4min** [comment]
> I'll update the default value for the `todo-tree.general.schemes` setting.

Thank you.


> Could you post your settings and an example notebook so I could try it out?

Yea sure!
[Notebook+Settings.zip](https://github.com/Gruntfuggly/todo-tree/files/7262776/Notebook%2BSettings.zip)
This is my todo-tree related settings (if you want the full settings, I'll send it) and a simple notebook file that includes these TAGs "fix", "todo", "bug" in two different cell.

As far as I know, Jupyter Notebooks are derived from the IPython project and support over 40 programming languages, including Python, R, Julia etc.

`.ipynb` (Interactive Python NoteBook) is a JSON file that gets rendered in editors to show it's content. Each notebook can contains a lot of cells.

There are two type of cell: "Markdown Cell" for documentation and "Code Cell".
Also there's an "Output Cell" attached below each code cell to display its result.

**Comment by Rchatru** [comment]
> Can you try adding `vscode-notebook-cell` to the `todo-tree.general.schemes` setting.
> 
> Please have a look at this issue and see if it helps: #548

Thank you very much. Now the tags are shown, but there is still the same problem as @dev4min indicates, only the tags present in the current cell are shown, and when you click on one of them, it does not take you to the line where it is located.

It would be a breakthrough if we manage to solve this.

**Comment by Rchatru** [comment]
In addition, there is another slight problem. As you can see in the gif, until you click on a cell, it seems that the plugin does not detect the label. It may seem like a minor problem, but this behaviour causes me to miss these tags when I review the code.
You can also see that when you put the mouse over TODO list, a line number is displayed that does not correspond to reality (1521), probably due to the internal construction of the "Jupyter Notebook" file, which is basically based on JSON. This makes that in reality, even if you have a list with TODOs, you cannot find them in the code, making this function useless.

I hope this information can be useful for you to fix the problem. Any information you need or any test you wish to perform, you can count on me.

![EDA_dudosos ipynb - Codigo-Doctorado - Visual Studio Code 2021-10-06 18-17-47_Trim](https://user-images.githubusercontent.com/38190230/136256218-9a1290ff-3988-4192-ab06-646e85291863.gif)

**Comment by Gruntfuggly** [comment]
Yes - it's finding the TODOs in the source for the notebook. To be honest, it might not be possible because the notebooks are a custom view - not a normal text editor.

I'm not sure the API for notebooks provides any way to interact with them at the moment. I'll try a few things out.

**Comment by stephenfrost** [comment]
I ran into this problem as well, today: tag not found in notebook unless I am in edit mode within the cell where the tag is. Odd thing is, just a few days ago, I think I did have some of these tags show up in the Todo tree, without editing I cell. I have already the default setting for notebook cells in my extension settings, as mentioned here and in other (closed) thread on this topic. Any progress on this topic since Oct 2021?

**Comment by pbeens** [comment]
It's finding #TODO in my Python code in Jupyter notebooks but not in markdown sections of the notebook. That's a bummer!

**Comment by schneipk** [comment]
I would really love to see this feature as I am working a lot in Jupyter Notebooks lately. I really hope @Gruntfuggly  finds the time to add support for jumping to the right lines, not only the right notebook. 
Great work! Thank you!  <3

**Comment by toiletbowlkuku** [comment]
> Can you try adding `vscode-notebook-cell` to the `todo-tree.general.schemes` setting.
> 
> Please have a look at this issue and see if it helps: #548

@Gruntfuggly is there a scheme that allows it to scan the entire notebook? As mentioned before this only scans the active cell.

**Comment by sadegh-pz** [comment]
@Gruntfuggly How can we help you fix this issue of scanning, showing, and jumping to a TODO comment of a .ipynb file?
